addon/registry.js

```diff
@@ -0,0 +1,57 @@
+let REGISTERED_RECEIVERS = {};
+
+export function registerReceiver(component) {
+  let componentName = getComponentRegistryKey(component);
+  REGISTERED_RECEIVERS[componentName] = component;
```

Comment:
I like the flexibility of keeping lists mapped to names:

```hbs
{{my-component sendToName="foo"}}
{{my-component sendToName="foo"}}
{{my-component sendToName="bar"}}
{{my-component sendToName="bar"}}
```

Thus "name" becomes a group instead of a key.